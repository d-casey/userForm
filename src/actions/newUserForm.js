import {
  RESET_USER_FORM,
  SUBMIT_NEW_USER_FORM
} from '../constants/actions'

export const submitNewUserForm = formFields => ({
  type: SUBMIT_NEW_USER_FORM,
  payload: formFields
})

export const resetUserForm = () => ({
  type: RESET_USER_FORM
})
