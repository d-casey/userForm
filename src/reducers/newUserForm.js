import {
  SUBMIT_NEW_USER_FORM,
  RESET_USER_FORM
} from '../constants/actions'

const initialState = {
  name: "",
  role: "",
  email: "",
  password: "",
  isSubmitted: false
}

export default function newUserForm(state = initialState, action) {
  switch (action.type) {
    case SUBMIT_NEW_USER_FORM:
      return {
        ...state,
        name: action.payload.name,
        role: action.payload.role,
        email: action.payload.email,
        password: action.payload.password,
        isSubmitted: true
      }

    case RESET_USER_FORM:
      return {
        ...state,
        name: "",
        role: "",
        email: "",
        password: "",
        isSubmitted: false
      }

    default:
      return state
  }
}
