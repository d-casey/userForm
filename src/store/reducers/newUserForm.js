import {
  UPDATE_USER_FIELDS,
  RESET_USER_FORM
} from '../constants/actions'

export const initialState = {
  name: "",
  role: "",
  email: "",
  password: "",
  isSubmitted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_USER_FIELDS:
      return {
        ...state,
        name: action.payload.name,
        role: action.payload.role,
        email: action.payload.email,
        password: action.payload.password,
        isSubmitted: true
      }

    case RESET_USER_FORM:
      return initialState

    default:
      return state
  }
}
