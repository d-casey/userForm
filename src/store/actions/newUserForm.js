import {
  RESET_USER_FORM,
  UPDATE_USER_FIELDS
} from '../constants/actions'

import { updateStage } from '../actions'

export const updateUserFields = formFields => ({
  type: UPDATE_USER_FIELDS,
  payload: formFields
})

export const submitNewUserForm = formFields => dispatch => {
  dispatch(updateUserFields(formFields))
  dispatch(updateStage(1))
}

export const resetUserForm = () => ({
  type: RESET_USER_FORM
})
