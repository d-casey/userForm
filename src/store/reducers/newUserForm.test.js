import {
  UPDATE_USER_FIELDS,
  RESET_USER_FORM
} from '../constants/actions'

import newUserForm from './newUserForm'

test('it updates the user fields correctly', () => {
  expect(newUserForm({},
      {
        type: UPDATE_USER_FIELDS,
        payload:
        {
          name: 'Severus Snape',
          role: 'Potions master',
          email: 'Sev@owlmail.com',
          password: 'Alw4ys',
          isSubmitted: true
        }
      }
    )
  )
  .toMatchSnapshot()
})

test('it resets the stage correctly', () => {
  expect(newUserForm({}, { type: RESET_USER_FORM })).toMatchSnapshot()
})

test('it returns the default state when nothing provided', () => {
  expect(newUserForm({}, { type: 'UNKNOWN' })).toMatchSnapshot()
})
