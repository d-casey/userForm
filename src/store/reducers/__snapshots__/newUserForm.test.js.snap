// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it resets the stage correctly 1`] = `
Object {
  "email": "",
  "isSubmitted": false,
  "name": "",
  "password": "",
  "role": "",
}
`;

exports[`it returns the default state when nothing provided 1`] = `Object {}`;

exports[`it updates the user fields correctly 1`] = `
Object {
  "email": "Sev@owlmail.com",
  "isSubmitted": true,
  "name": "Severus Snape",
  "password": "Alw4ys",
  "role": "Potions master",
}
`;
